services/scabbard/libscabbard/src/service/shared.rs

```diff
@@ -95,15 +106,84 @@ impl ScabbardShared {
         &self.coordinator_service_id
     }
 
-    pub fn add_batch_to_queue(&mut self, batch: BatchPair) {
+    /// Returns the number of pending batches
+    pub fn pending_batches_count(&self) -> usize {
+        self.batch_queue.len()
+    }
+
+    /// set whether we are accepting new batches
+    pub fn set_accepting_batches(&mut self, accepting: bool) {
```

Comment:
If this does not need to be pub it should be removed and the boolean accessed directly.

Reply:
It does need to be pub, non-coordinators use this when receiving the TOO_MANY_REQUESTS/ACCEPTING_REQUESTS message from the coordinators